### DISCUSSION TITLE
Request body based routing

### DISCUSSION THREAD

**Discussion by tanshunyuan** [Kong/kong#10455] [Help] [answered]
Let's say I have an API Gateway and a server. 

Server:
* /plans/single
* /plans/multiple

API Gateway 
* /plans

An incoming request to the API Gateway will probably look like:

```
POST /plans

Request Body
{
  type: 'SINGLE'
}
```
Based on the request body will it be possible for kong to invoke the `/plans/single` endpoint that lives on the server?

**Comment by shettyh** [comment, marked as answer]
I don't think it is possible out of the box. But you can write a plugin to do that

1. Doc for writing custom plugin : https://docs.konghq.com/gateway/latest/plugin-development/
2. Reading body in plugin using Kong pdk method: https://docs.konghq.com/gateway/latest/plugin-development/pdk/kong.request/#kongrequestget_bodymimetype-max_args
3. Change the request path using PDK based on the body: https://docs.konghq.com/gateway/latest/plugin-development/pdk/kong.service.request/#kongservicerequestset_pathpath

**Reply by tanshunyuan** [reply]
alright thank you!
